src/components/swap/ConfirmSwapModal.tsx

```diff
@@ -1,39 +1,172 @@
 import { Trans } from '@lingui/macro'
-import { sendAnalyticsEvent, Trace } from '@uniswap/analytics'
-import { InterfaceModalName, SwapEventName, SwapPriceUpdateUserResponse } from '@uniswap/analytics-events'
+import { sendAnalyticsEvent, Trace, useTrace } from '@uniswap/analytics'
+import {
+  InterfaceEventName,
+  InterfaceModalName,
+  SwapEventName,
+  SwapPriceUpdateUserResponse,
+} from '@uniswap/analytics-events'
 import { Percent } from '@uniswap/sdk-core'
+import { useWeb3React } from '@web3-react/core'
+import Modal from 'components/Modal'
+import { useMaxAmountIn } from 'hooks/useMaxAmountIn'
+import { Allowance, AllowanceState } from 'hooks/usePermit2Allowance'
+import usePrevious from 'hooks/usePrevious'
 import { getPriceUpdateBasisPoints } from 'lib/utils/analytics'
-import { ReactNode, useCallback, useEffect, useMemo, useState } from 'react'
+import { ReactNode, useCallback, useEffect, useState } from 'react'
 import { InterfaceTrade } from 'state/routing/types'
+import { useIsTransactionConfirmed } from 'state/transactions/hooks'
+import invariant from 'tiny-invariant'
 import { formatSwapPriceUpdatedEventProperties } from 'utils/loggingFormatters'
+import { didUserReject } from 'utils/swapErrorToUserReadableMessage'
 import { tradeMeaningfullyDiffers } from 'utils/tradeMeaningFullyDiffer'
 
-import TransactionConfirmationModal, {
-  ConfirmationModalContent,
-  TransactionErrorContent,
-} from '../TransactionConfirmationModal'
+import { ConfirmationModalContent } from '../TransactionConfirmationModal'
+import {
+  ErrorModalContent,
+  PendingConfirmModalState,
+  PendingModalContent,
+  PendingModalError,
+} from './PendingModalContent'
 import SwapModalFooter from './SwapModalFooter'
 import SwapModalHeader from './SwapModalHeader'
 
+export enum ConfirmModalState {
+  REVIEWING,
+  APPROVING_TOKEN,
+  PERMITTING,
+  PENDING_CONFIRMATION,
+}
+
+function isAllowing(confirmModalState: ConfirmModalState) {
```

Comment:
i can rename to something like `isInApprovePhase` ? it's just a shorthand for checking if the current state is one of two values. i also use the function more in stacked diffs

Reply:
If the name would be `isInApprovalPhase`, I would expect the second argument to be `||` instead of `&&`. Reading in English: either is approving or is approved. So yeah, I guess I just need to understand the reason behind conditional!

Reply:
this specific conditional is checking if we need to move the `confirmModalState` past the "approving" phase, based on whether `allowance` has updated. so this is expected to be &&  - it means "we are in the approval phase, but it's just ended because the final approval we needed has finalized"

Reply:
discussed offline and i added a comment explaining this here